core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/util/SigningAlgorithm.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.internal.util;
+
+import java.security.NoSuchAlgorithmException;
+import javax.crypto.Mac;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+@SdkInternalApi
+public enum SigningAlgorithm {
+
+    HmacSHA256;
```

Comment:
should we take this opportunity to rename the enum to fit the usual `UPPERCASE_UNDERSCORE` naming convention?